src/core/server/elasticsearch/errors.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Boom from 'boom';
+import { get } from 'lodash';
+
+const code = Symbol('ElasticsearchError');
+
+const CODE_NOT_AUTHORIZED = 'Elasticsearch/notAuthorized';
+
+export interface ElasticsearchError extends Boom {
+  [code]?: string;
```

Comment:
__nit:__ `readonly [code]?: string;`?

Reply:
that complicates a signature as we mutate object to assign this value

Reply:
Hmm, weird, I thought we should treat error as `ElasticsearchError` only after we set this `code`. But yeah, it depends on how you implement it.